### ISSUE TITLE
Microsoft.Graph.Authentication Invoke-MgGraphRequest is broken

### ISSUE THREAD

**Issue by tjgruber** [microsoftgraph/msgraph-sdk-powershell#2495]
**Describe the bug**

Now getting failures with Graph API requests since Dec. 13th after ~3pm PST.

I think this broke it: https://github.com/microsoftgraph/msgraph-sdk-powershell/pull/2455/files
Possibly other Microsoft.Graph.Authentication issues, as I haven't dug much deeper into that module's changes.

**To Reproduce**
1. Delete ALL powershell modules from PS modules folders
2. Ran `Install-Module -Name Microsoft.Graph.Authentication -RequiredVersion 2.10.0`
3. Tested the following: `$AllIntuneDevices = Get-GraphRequest -URI 'https://graph.microsoft.com/beta/deviceManagement/managedDevices?$top=999'`
4. Above ran successfully.
5. Repeated Step #1.
6. Ran `Install-Module -Name Microsoft.Graph.Authentication -RequiredVersion 2.11.0`
7. Repeated Step #3.
8. Failed. `"code":"BadRequest","message":"Resource not found for the segment 'managedDevices?$top=999'`

**Expected behavior**
> I expected the API call to be successful as it has always been in the past.

**Module Version**
2.11.0 Microsoft.Graph.Authentication

**Environment Data**
PSVersion                      7.3.10
PSEdition                      Core
GitCommitId                    7.3.10
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

**Additional context**
I first noticed this in GitHub actions. 
After checking the [change log](https://github.com/actions/runner-images/commit/eed37c2c66895b95f6e8d47f94c053a924e3f90c) for the "windows-latest" github actions runner, I discovered Microsoft.Graph was updated. Through various breadcrumbs, led me to https://github.com/microsoftgraph/msgraph-sdk-powershell/pull/2455/files.

**Comment by timayabi2020** [comment]
Hi @tjgruber I was not able to reproduce the issue because I'm getting a success with the said SDK version. Can you please share your debug output?
<img width="911" alt="image" src="https://github.com/microsoftgraph/msgraph-sdk-powershell/assets/10947120/c0d4f0bf-09f6-4900-a413-aafa2ae3f418">

**Comment by tjgruber** [comment]
> Hi @tjgruber I was not able to reproduce the issue because I'm getting a success with the said SDK version. Can you please share your debug output? <img alt="image" width="911" src="https://private-user-images.githubusercontent.com/10947120/291521055-c0d4f0bf-09f6-4900-a413-aafa2ae3f418.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMwMzAzMDcsIm5iZiI6MTcwMzAzMDAwNywicGF0aCI6Ii8xMDk0NzEyMC8yOTE1MjEwNTUtYzBkNGYwYmYtMDlmNi00OTAwLWE0MTMtYWFmYTJhZTNmNDE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE5VDIzNTMyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNhZjIxYWU0ZjI5MTM5ODVkMjk1M2JjMjY1OTliZjA5YWU5OTYwMjk3NDQ0MjA5OGFkZjllZmFiZGE4YjExNmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.eCAawo7XAKJOWZ69n3vCOZNnsci1G5hmJZo711EbBpE">

Yes, it's happening on my device in a regular PowerShell window, as well as in the "windows-latest" GitHub Actions runner. I made a quick example. 

Again, I know this issue is already fixed in 2.11.1. The issue is that the windows-latest runner needs to be updated to use the 2.11.1 module of Microsoft.Graph.Authentication.

Local computer:
![image](https://github.com/microsoftgraph/msgraph-sdk-powershell/assets/15575143/02635d8a-ee23-4fed-85c2-3f91e2dc5abc)

GitHub Actions "windows-latest" runner: (same commands as local, but without -debug)
![image](https://github.com/microsoftgraph/msgraph-sdk-powershell/assets/15575143/24de0efb-cbd1-4faf-b7eb-d25a525a7a5c)

**Comment by tjgruber** [comment]
This now appears to be fixed in the GitHub runners.
